library/src/main/java/com/parse/internal/signpost/basic/HttpURLConnectionClient.java

```diff
@@ -0,0 +1,56 @@
+package com.parse.internal.signpost.basic;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Singleton class to provide HttpURLConnection, using native HttpURLConnection or OkHttp
+ */
+public class HttpURLConnectionClient {
+    private boolean isUsingOkHttp = false;
+    private Object okUrlFactory = null;
+    private Method okUrlFactoryOpen = null;
+
+    protected HttpURLConnectionClient() {
+        try {
+            Class okHttpClientClass = Class.forName("com.squareup.okhttp.OkHttpClient");
+            Object okHttpClient = okHttpClientClass.getConstructor().newInstance();
+            Class okUrlFactoryClass = Class.forName("com.squareup.okhttp.OkUrlFactory");
+            okUrlFactory = okUrlFactoryClass.getConstructor(okHttpClientClass).newInstance(okHttpClient);
+            okUrlFactoryOpen = okUrlFactoryClass.getMethod("open", URL.class);
+            isUsingOkHttp = true;
+        } catch (Exception e) {
+            isUsingOkHttp = false;
+        }
+    }
+
+    private static HttpURLConnectionClient instance = null;
+    public static HttpURLConnectionClient getInstance() {
+        if (instance == null) {
+            synchronized (HttpURLConnectionClient.class) {
+                if (instance == null) {
+                    instance = new HttpURLConnectionClient();
+                }
+            }
+        }
+        return instance;
+    }
```

Comment:
We're not the biggest fans of static globals, could you move the reference to a instance on `Twitter` that get's passed into `DefaultOAuthProvider`? I'm ok with removing the global reference, but using a static factory `newInstance()` instead of `getInstance()`.